controllers/authenticationController.js

```diff
@@ -0,0 +1,97 @@
+    console.log(req.body);
+
+    const { name, password, email, mobile, address } = req.body;
+
+    bcrypt.hash(password, 12, async function (err, hash) {
+
+        if (err) {
+            return res.status(403).json({
+                status: 'error',
+                error: err,
+                message: err.message
+            });
+        }
+
+        const user = await User.create({
+            name,
+            password: hash,
+            email,
+            mobile,
+            address
+        },
+            function (err, model) {
+                if (err) {
+                    return res.status(403).json({
+                        status: 'error',
+                        error: err,
+                        message: err.message
+                    });
+                }
+                delete model._doc.password;
+                return res.status(200).json({
+                    status: 'success',
+                    data: model
+                });
+            });
+    });
+});
+
+exports.login = catchAsync(async (req, res, next) => {
+    console.log(req.body);
+
+    const { email, password } = req.body;
+
+    User.findOne({ email: email }, function (err, model) {
+        if (err) {
+            console.log(err);
+            return res.status(403).json({
+                status: 'error',
+                error: err,
+                message: err.message
+            });
+        }
+        console.log(model)
+
+        if (model) {
+            bcrypt.compare(password, model.password)
+                .then(matched => {
+                    if (matched) {
+                        const token = jwt.sign({ _id: model._id }, process.env.JWT_KEY);
+                        delete model._doc.password;
+                        return res.status(200).json({
+                            token,
+                            status: 'success',
+                            user: {
+                                ...model._doc
+                            },
+                            message: "Successfully Logged In"
+                        });
+                    } else {
+                        return res.status(403).json({
+                            status: 'error',
+                            error: "User Found but Password is wrong",
+                            message: "Invalid Email or Password"
+                        });
+                    }
+                }).catch(err => {
+                    return res.status(400).json({
+                        status: 'error',
+                        error: "Something wrong! Maybe password is null, or JWT_KEY is null",
+                        message: err.message
+                    });
+                });
+        } else {
+            return res.status(404).json({
+                status: 'error',
+                error: "No User Found with entered Email",
+                message: "Invalid Email or Password"
+            });
+        }
```

Comment:
Instead of .then and callback approach resolve promise by await. Current code is unnecessarily increasing indentation. If you use await catchAsync will catch the error, you don't have to handle the error.

Reply:
I didn't get the logic of not handling error by ourselves. Please explain this. Because, as of now I can see that, If I will try to call an API that doesn't exist, like "/api/hello", I didn't get any response. And I think it is wrong. At least we should reply to the API caller that "The API which you are trying, doesn't exist." Let me know your views.

Reply:
In config.env file add NODE_ENV="production"

![Screenshot 2021-01-05 094506](https://user-images.githubusercontent.com/50537259/103605876-0c1faf80-4f3b-11eb-8486-fbf8aad1707a.jpg)

or NODE_ENV="dvelopment"
![Screenshot 2021-01-05 094635](https://user-images.githubusercontent.com/50537259/103606001-5c970d00-4f3b-11eb-84c3-d446197e462a.jpg)